plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/HttpResponseSender.java

```diff
@@ -26,5 +27,5 @@
  */
 public interface HttpResponseSender
 {
-  void send(Response response, HttpServletResponse httpServletResponse) throws ServletException, IOException;
+  void send(Response response, HttpServletRequest httpRequest, HttpServletResponse httpServletResponse) throws ServletException, IOException;
```

Comment:
FTR I think the handling of HEAD in the sender is fine for this now.  My comment above was only to say use the Request not the HttpServerRequest.  So I'd change ResponseSender api to send(Request,Response,HttpServletResponse) so that the sender impl has all the information from request and response to render the servlet response.